integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/IntegrasjonspunktApplication.java

```diff
@@ -45,15 +48,33 @@ public static void main(String[] args) {
                 return;
             }
 
-            new SpringApplicationBuilder(IntegrasjonspunktApplication.class)
+            ConfigurableApplicationContext context = new SpringApplicationBuilder(IntegrasjonspunktApplication.class)
                     .initializers(new SpringCloudProtocolResolver())
                     .run(args);
+            checkNtpSync(context);
 
         } catch (SecurityException se) {
             logMissingJCE(se);
         }
     }
 
+    private static void checkNtpSync(ConfigurableApplicationContext context) {
+        String host = context.getBean(IntegrasjonspunktProperties.class).getNtpHost();
+        try {
+            NTPClient client = new NTPClient(host);
+            long offset = client.getOffset();
+            if (Math.abs(offset) > 5000) {
+                String errorStr = String.format("Startup failed. Offset from NTP host %s was more than 5 seconds (%sms). Adjust local clock and try again.", host, offset);
+                log.error(errorStr);
+                String stars = "\n**************************\n";
+                System.out.println(stars+errorStr+stars);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS106)